pkg/controller/controller_instance.go

```diff
@@ -329,6 +353,86 @@ func (c *controller) initOrphanMitigationCondition(instance *v1beta1.ServiceInst
 	return false, nil
 }
 
+// setNextProvisionRetryTime adds the instance name to the map recording the next earliest time to
+// retry provisioning
+func (c *controller) setNextProvisionRetryTime(instance *v1beta1.ServiceInstance) {
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(instance)
+	if err != nil {
+		glog.Errorf("Couldn't get key for object %+v: %v", instance, err)
+		return
+	}
+	duration := c.provisionRetryQueue.rateLimiter.When(key)
+	c.provisionRetryQueue.mutex.Lock()
+	c.provisionRetryQueue.retryTime[key] = time.Now().Add(duration)
+	glog.V(4).Infof("provisionRetry for %s after %v", key, duration)
+	c.provisionRetryQueue.mutex.Unlock()
+}
+
+//  delayProvisionRetry returns a duration which should be observed before attempting to provision
+func (c *controller) getDelayForProvisionRetry(instance *v1beta1.ServiceInstance) time.Duration {
+	c.purgeExpiredRetryEntries()
```

Comment:
`get...` in the method name suggests that this method should not have any side effects.
I would prefer to explicitly invoke `purgeExpiredRetryEntries` when needed, rather then doing it here hidden from the user.